gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -173,9 +177,14 @@ private void performRequest(
             future = future.transform(response -> recordBackendForQueryId(request, response), executor);
         }
 
+        Map<String, String> extraHeaders = new HashMap<>();
```

Comment:
The mutable maps should be avoided.

This could be just:
`Map<String, String> extraHeaders = includeClusterInfoInResponse ? ImmutableMap.of(CLUSTER_HOST_HEADER, remoteUri.getHost()) : ImmutableMap.of()`